include/deal.II/base/partitioner.templates.h

```diff
@@ -107,33 +107,37 @@ namespace Utilities
         }
 
       Number *temp_array_ptr = temporary_storage.data();
-#    if defined(DEAL_II_WITH_CUDA) && defined(DEAL_II_MPI_WITH_CUDA_SUPPORT)
-      // When using CUDAs-aware MPI, the set of local indices that are ghosts
+#    if defined(DEAL_II_MPI_WITH_DEVICE_SUPPORT)
+      // When using device-aware MPI, the set of local indices that are ghosts
       // indices on other processors is expanded in arrays. This is for
       // performance reasons as this can significantly decrease the number of
       // kernel launched. The indices are expanded the first time the function
       // is called.
-      if ((std::is_same<MemorySpaceType, MemorySpace::CUDA>::value) &&
+      if ((std::is_same<MemorySpaceType, MemorySpace::Default>::value) &&
           (import_indices_plain_dev.size() == 0))
         initialize_import_indices_plain_dev();
 #    endif
 
       for (unsigned int i = 0; i < n_import_targets; ++i)
         {
-#    if defined(DEAL_II_WITH_CUDA) && defined(DEAL_II_MPI_WITH_CUDA_SUPPORT)
-          if (std::is_same<MemorySpaceType, MemorySpace::CUDA>::value)
+#    if defined(DEAL_II_MPI_WITH_DEVICE_SUPPORT)
+          if (std::is_same<MemorySpaceType, MemorySpace::Default>::value)
             {
-              const auto chunk_size = import_indices_plain_dev[i].second;
-              const int  n_blocks =
-                1 + chunk_size / (::dealii::CUDAWrappers::chunk_size *
-                                  ::dealii::CUDAWrappers::block_size);
-              ::dealii::LinearAlgebra::CUDAWrappers::kernel::
-                gather<<<n_blocks, ::dealii::CUDAWrappers::block_size>>>(
-                  temp_array_ptr,
-                  import_indices_plain_dev[i].first.get(),
-                  locally_owned_array.data(),
-                  chunk_size);
-              cudaDeviceSynchronize();
+              const auto chunk_size = import_indices_plain_dev[i].size();
+              using IndexType       = decltype(chunk_size);
+
+              MemorySpace::Default::kokkos_space::execution_space exec;
+              Kokkos::parallel_for(
+                "fill temp_array_ptr",
+                Kokkos::RangePolicy<
+                  MemorySpace::Default::kokkos_space::execution_space>(
+                  exec, 0, chunk_size),
+                KOKKOS_LAMBDA(IndexType idx) {
+                  temp_array_ptr[idx] =
+                    locally_owned_array
+                      .data()[import_indices_plain_dev[i](idx)];
+                });
```

Comment:
I am surprised that you can call `import_indices_plain_dev[i]` on the device because `import_indices_plain_dev` is a `std::vector` on the host.